extensions/amp-story/1.0/amp-story-grid-layer.js

```diff
@@ -84,24 +87,33 @@ export class AmpStoryGridLayer extends AmpStoryBaseLayer {
     super(element);
 
     /** @private {?boolean} */
-    this.isFirstPage_ = null;
+    this.isActivePage_ = null;
 
     /** @private {?{horiz: number, vert: number}} */
     this.aspectRatio_ = null;
   }
 
   /**
-   * Returns true if a child of the first page.
+   * Returns true if the page should be prerendered (for being an active page or first page)
    * @return {boolean}
    */
-  isFirstPage() {
-    if (this.isFirstPage_ === null) {
-      this.isFirstPage_ = matches(
-        this.element,
-        'amp-story-page:first-of-type amp-story-grid-layer'
-      );
+  isPrerenderActivePage() {
+    if (this.isActivePage_ != null) {
+      return this.isActivePage_;
     }
-    return this.isFirstPage_;
+    const hashId = parseQueryString(this.win.location.href)['page'];
+    let selector = 'amp-story-page:first-of-type';
+    if (hashId) {
+      selector += `, amp-story-page#${escapeCssSelectorIdent(hashId)}`;
+    }
+    const parentPage = closest(
+      this.element,
+      (el) => el.tagName.toLowerCase() === 'amp-story-page'
+    );
```

Comment:
It's a micro optimization but in the prerender case we can be extra cautious: given that `amp-story-grid-layer` HAS to be a direct descendant of `amp-story-page`, you can use `this.element.parentElement`